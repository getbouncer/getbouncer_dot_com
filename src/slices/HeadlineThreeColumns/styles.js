import styled from '@emotion/styled'
import { breakpoints, theme } from '~styles'

export const Headline = styled.div`
  margin-bottom: 64px;
  text-align: center;

  h2 {
    background: ${theme.textGradient};
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    display: inline-block;
    font-size: 3.5rem;
    line-height: 4.5rem;

    @media ${breakpoints.smartphone} {
      font-size: 2.85rem;
      line-height: 1.1;
      text-align: left;
    }
  }
`

export const Row = styled.ul`
  display: flex;
  justify-content: space-between;

  @media ${breakpoints.smartphone} {
    display: block;
  }
`

export const Item = styled.li`
  flex: 0 0 28%;

  @media ${breakpoints.smartphone} {
    margin-bottom: 50px;
    width: 100%;
  }
`

export const Icon = styled.img`
  height: 72px;
`

export const ItemHeadline = styled.h3`
  margin: 24px 0 18px;
`
