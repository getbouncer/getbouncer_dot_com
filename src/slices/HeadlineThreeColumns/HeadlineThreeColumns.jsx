import React from 'react'
import PropTypes from 'prop-types'
import Container from '~globals/Container'
import { Headline, Row, Item, Icon, ItemHeadline } from './styles'

const HeadlineThreeColumns = ({ input }) => {
  const { items, primary } = input

  return (
    <section>
      <Container>
        <Headline dangerouslySetInnerHTML={{ __html: primary.headline.html }} />
        <Row>
          {items.map(item => (
            <Item key={item.item_title}>
              <Icon src={item.item_image.url} alt={item.item_image.alt} />
              <ItemHeadline>{item.item_title}</ItemHeadline>
              <div dangerouslySetInnerHTML={{ __html: item.item_description.html }} />
            </Item>
          ))}
        </Row>
      </Container>
    </section>
  )
}

HeadlineThreeColumns.propTypes = {
  input: PropTypes.shape().isRequired,
}

export default HeadlineThreeColumns
