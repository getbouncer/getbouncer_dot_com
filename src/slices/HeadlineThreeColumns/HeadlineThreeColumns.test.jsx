import React from 'react'
import renderer from 'react-test-renderer'
import HeadlineThreeColumns from './HeadlineThreeColumns'

/* eslint-disable no-undef */

describe('HeadlineThreeColumns', () => {
  it('renders correctly', () => {
    const data = {
      primary: {
        headline: {
          html: '<h1>headline</h1>',
        },
      },
      items: [
        {
          item_title: 'item 1 title',
          item_image: {
            alt: 'item 1 image',
            url: 'https://www.example.com/item_1_image.png',
          },
          item_description: {
            html: '<p>item description</p>',
          },
        },
        {
          item_title: 'item 2 title',
          item_image: {
            alt: 'item 2 image',
            url: 'https://www.example.com/item_2_image.png',
          },
          item_description: {
            html: '<p>item description</p>',
          },
        },
      ],
    }
    const tree = renderer
      .create(<HeadlineThreeColumns input={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
