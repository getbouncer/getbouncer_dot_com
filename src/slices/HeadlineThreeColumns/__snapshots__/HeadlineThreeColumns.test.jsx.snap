// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadlineThreeColumns renders correctly 1`] = `
<section>
  <div
    className="css-1fywgw4"
  >
    <div
      className="css-14ndujd"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h1>headline</h1>",
        }
      }
    />
    <ul
      className="css-2mxeky"
    >
      <li
        className="css-iylok8"
      >
        <img
          alt="item 1 image"
          className="css-1mhm7c2"
          src="https://www.example.com/item_1_image.png"
        />
        <h3
          className="css-5ooxch"
        >
          item 1 title
        </h3>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>item description</p>",
            }
          }
        />
      </li>
      <li
        className="css-iylok8"
      >
        <img
          alt="item 2 image"
          className="css-1mhm7c2"
          src="https://www.example.com/item_2_image.png"
        />
        <h3
          className="css-5ooxch"
        >
          item 2 title
        </h3>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>item description</p>",
            }
          }
        />
      </li>
    </ul>
  </div>
</section>
`;
